### PR TITLE
fix(multimodal): handle the case when 'input_ids' is None

### PR DESCRIPTION
**Pull request by JiaoPL**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation of this PR and the goal you want to achieve through this PR.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here and update the documentation.

## Checklist

**Before PR**:

- [ ] Pre-commit or other linting tools are used to fix the potential lint issues.
- [ ] Bug fixes are fully covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [ ] The documentation has been modified accordingly, like docstring or example tutorials.

**After PR**:

- [ ] If the modification has potential influence on downstream or other related projects, this PR should be tested with those projects.
- [ ] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: internlm/data/build_dataloader.py [hunk 1]
```diff
@@ -33,7 +33,7 @@ def get_tokenized_train_loader_items(data_cfg):
         if data_cfg.get("is_multimodal", False):
             image_token_size = int(data_cfg.image_size // data_cfg.patch_size) ** 2
             train_ds = RandomDatasetMultimodal(
-                num_samples=10000,
+                num_samples=100000,
```
Comment (JiaoPL) [inline internlm/data/build_dataloader.py:36]: 这个会导致 build dataset时特别慢，但10000samples在64+ gpus上可能不太够，要改回去吗？

Comment (huangting4201) [reply internlm/data/build_dataloader.py:36]: 都行，我看llm的num_samples=1000000，也挺慢吧

Comment (sunpengsdu) [reply internlm/data/build_dataloader.py:36]: 就 100000 吧先

Comment (JiaoPL) [reply internlm/data/build_dataloader.py:36]: 好的

File: internlm/model/modeling_internlm.py [hunk 2]
```diff
@@ -358,7 +358,7 @@ def __init__(
 
     def forward(self, hidden_states=None, cu_seqlens=None, input_ids=None, indexes=None, inference_params=None):
         # attention_mask: compute attention on the places where the value is 1
-        if hasattr(self, "embedding"):
+        if hasattr(self, "embedding") and input_ids is not None:
```
Comment (sunpengsdu) [inline internlm/model/modeling_internlm.py:361]: 更好的办法是main函数那里如果有了空的input id直接assert err掉？

Comment (huangting4201) [reply internlm/model/modeling_internlm.py:361]: 龙佬加这个是为了跟vit一起用，允许hidden_states不为None，而input_ids为None的情况

**Review by sunpengsdu — APPROVED** [review]

**Pull request merged**